projects/igniteui-angular/src/lib/services/exporter-common/export-utilities.ts

```diff
@@ -68,7 +68,9 @@ export class ExportUtilities {
                               .replace(/</g, '&lt;')
                               .replace(/>/g, '&gt;')
                               .replace(/"/g, '&quot;')
-                              .replace(/'/g, '&apos;');
+                              .replace(/'/g, '&apos;')
+                              // Bug #14944 - Replace the not supported null character (\u0000, \x00) with a space
+                              .replace(/\0/g, '&#32;');
```

Comment:
Just cuz `\0` looked odd to me - I checked. It's technically a [legacy escape](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_escape#0), so perhaps we should avoid using it and just have `\x00` instead.
PS: Also still thinking the null char should be empty space (see [wiki](https://en.wikipedia.org/wiki/Null_character)), but really don't feel too strongly about it.

Reply:
IMO, we shouldn't be displaying it at all. I don't particularly mind the current implementation though.
It's almost tempting to say that we should cut off the string at the null byte as it is intended (why would you even be pushing null bytes into there unless you're trying to do something wonky?), but that's a whole separate can of worms...

Reply:
> Just cuz `\0` looked odd to me - I checked. It's technically a [legacy escape](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_escape#0), so perhaps we should avoid using it and just have `\x00` instead. 
Not sure what you mean? We don't use it, we just check if it somehow ended up as part of the cell value and make sure to replace it with something that won't break the XML.  
PS: Also still thinking the null char should be empty space (see [wiki](https://en.wikipedia.org/wiki/Null_character)), but really don't feel too strongly about it.
If I understand correctly, you prefer the non-breaking space (&#160;), right?

Reply:
> Just cuz `\0` looked odd to me - I checked. It's technically a [legacy escape](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_escape#0), so perhaps we should avoid using it and just have `\x00` instead.

Not sure what you mean? We don't use it, we just check if it somehow ended up as part of the cell value and make sure to replace it with something that won't break the XML.

Reply:
> PS: Also still thinking the null char should be empty space (see [wiki](https://en.wikipedia.org/wiki/Null_character)), but really don't feel too strongly about it.

If I understand correctly, you prefer the non-breaking space (&#160;), right?

Reply:
> It's almost tempting to say that we should cut off the string at the null byte as it is intended (why would you even be pushing null bytes into there unless you're trying to do something wonky?), but that's a whole separate can of worms...

Agreed, it's very tempting to cut it off but I sincerely doubt they would expect that. I'm assuming this is data coming from a db and those characters have accidentally ended up as cell values.

Reply:
> Not sure what you mean? We don't use it, we just check if it somehow ended up as part of the cell value and make sure to replace it with something that won't break the XML.

@gedinakova The comment in on the line with the regex using octal escape (`\0`) which is technically in the deprecated and obsolete page (see link). It might never be removed, however just saying we could use the hex escape instead :)

Also
> If I understand correctly, you prefer the non-breaking space (`&#160;`), right?

Nope, quite literally empty, as in replace with `''`.